plugins/raiseStat/raiseStat.pl

```diff
@@ -0,0 +1,185 @@
+package raiseStat;
+
+use strict;
+use Plugins;
+use Utils;
+use Globals qw(%config $net $char $messageSender);
+use Log qw(message debug error);
+use Network::PacketParser qw(STATUS_STR STATUS_AGI STATUS_VIT STATUS_INT STATUS_DEX STATUS_LUK);
+
+
+Plugins::register('raiseStat', 'automatically raise character stats', \&on_unload);
+
+################################################################
+#  Hooks used to activate the plugin during initialization
+#  and on config change events.
+my $base_hooks = Plugins::addHooks(
+	['start3',        \&checkConfig],
+	['postloadfiles', \&checkConfig],
+    ['configModify',  \&on_configModify]
+   );
+
+my @stats_to_add;
+my $active_hooks;
+my $adding_hook;
+my $next_stat;
+my $status;
+my $timeout = { time => 0, timeout => 1 };
+
+use constant {
+	INACTIVE => 0,
+	ACTIVE => 1,
+	ADDING => 2
+};
+
+sub on_unload {
+   Plugins::delHook($base_hooks);
+   changeStatus(INACTIVE);
+   message "raiseStat plugin unloading or reloading\n", 'success';
+}
+
+################################################################
+#  changeStatus() is the function responsible for adding
+#  and deleting hooks when changing plugin status.
+#  During status == 0 the plugin will be inactive and won't
+#  look for opportunities to raise stats.
+#  During status == 1 the plugin will be active and will
+#  have 'speculative' hooks added to try to look for
+#  opportunities to raise stats.
+#  During status == 2 the plugin will be active and will
+#  have 'AI_pre' hook active, on each AI call the plugin
+#  will try to raise stats if possible.
+sub changeStatus {
+	my $new_status = shift;
+	Plugins::delHook($active_hooks) if ($status == ACTIVE);
```

Comment:
'changeStatus' makes sure '$adding_hook' will only be active during 'ADDING' and '$active_hooks' will only be active during 'ACTIVE', so deleting them this way is okay.

And the conditions are different, first one is 'if ($status == ACTIVE)' and second one is 'if ($status == ADDING)'.